src/bp-forums/admin/forums.php

```diff
@@ -80,6 +80,36 @@ private function setup_actions() {
 
 			// Set forum states
 			add_filter( 'display_post_states', array( $this, 'bbp_set_hidden_forum_states' ), 10, 2 );
+
+			// Filter post parent for forum type post.
+			add_filter( 'wp_insert_post_parent', array( $this, 'forum_parent' ), 10, 3 );
+		}
+
+		/**
+		 * Permission for forum parent, If not then return the current parent.
+		 * You can not make the forum a child forum when it's associated with any group.
+		 *
+		 * @since BuddyBoss 1.7.6
+		 *
+		 * @param init  $post_parent post parent.
+		 * @param init  $post_ID     post ID.
+		 * @param array $new_postarr submited post data.
+		 *
+		 * @return init
+		 */
+		public function forum_parent( $post_parent, $post_ID, $new_postarr ) {
```

Comment:
move this function to the end of the file.

Reply:
Move the function at the end of the file.
https://github.com/buddyboss/buddyboss-platform/pull/2117/commits/b2c663fb81f0630d02690c0ac5064d62ff9f8387